graphite/graphite.go

```diff
@@ -0,0 +1,128 @@
+package graphite
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	// ErrBindAddressRequired is returned when starting the Server
+	// without a TCP or UDP listening address.
+	ErrBindAddressRequired = errors.New("bind address required")
+
+	// ErrServerClosed return when closing an already closed graphite server.
+	ErrServerClosed = errors.New("server already closed")
+
+	// ErrDatabaseNotSpecified retuned when no database was specified in the config file
+	ErrDatabaseNotSpecified = errors.New("database was not specified in config")
+
+	// ErrServerNotSpecified returned when Server is not specified.
+	ErrServerNotSpecified = errors.New("server not present")
+)
+
+// SeriesWriter defines the interface for the destination of the data.
+type SeriesWriter interface {
+	WriteSeries(database, retentionPolicy, name string, tags map[string]string, timestamp time.Time, values map[string]interface{}) error
+}
+
+// Parser encapulates a Graphite Parser.
+type Parser struct {
+	Separator   string
+	LastEnabled bool
+}
+
+// NewParser returns a GraphiteParser instance.
+func NewParser() *Parser {
+	p := Parser{}
+	p.Separator = "."
+	return &p
+}
+
+// Parse performs Graphite parsing of a single line.
+func (p *Parser) Parse(line string) (*metric, error) {
```

Comment:
Good catch -- that is not good. These methods were unexported at one point, but late last might I exported them so the unit tests could run. Let me just make `metric` exported for now too. I do agree that a general system-wide metric type could be the way to go.